pkg/operator/ceph/config/store.go

```diff
@@ -152,7 +158,23 @@ func (s *Store) createOrUpdateMonHostSecrets(clusterInfo *cephconfig.ClusterInfo
 	members := make([]string, len(clusterInfo.Monitors))
 	i := 0
 	for _, m := range clusterInfo.Monitors {
-		hosts[i] = m.Endpoint
+		monIP := cephutil.GetIPFromEndpoint(m.Endpoint)
+
+		// This tries to detect the current port if the mon already exists
+		// This basically handles the transtion between monitors running on 6790 to msgr2
+		// So whatever the previous monitor port was we keep it
+		currentMonPort := cephutil.GetPortFromEndpoint(m.Endpoint)
+
+		monPorts := [2]string{strconv.Itoa(int(Msgr2port)), strconv.Itoa(int(currentMonPort))}
+		msgr2Endpoint := net.JoinHostPort(monIP, monPorts[0])
+		msgr1Endpoint := net.JoinHostPort(monIP, monPorts[1])
+
+		if cephv1.VersionAtLeast(clusterInfo.CephVersionName, cephv1.Nautilus) {
```

Comment:
I think this check is not right.

What matters here is the version of ceph installed in the container where we'll consume this config file. We need to test whether the locally installed ceph is at least Nautilus, instead of testing vs. the clusterInfo.CephVersionName (which pertains to the version in the ceph daemon containers).

Reply:
Since this is a very short term problem I'd say let's just assume the base image is mimic. We can re-enable this check when we change the base image in a few weeks.

Reply:
That's probably fine as a workaround. Won't that prevent the daemons from actually using msgr2 though? You'll effectively be disabling it in the clients, no?

Reply:
Right, the daemons could still expose the new ports, even if the client has msgr2 disabled where it's not supported yet.

Reply:
As discussed in container huddle, Nautilus RC is expected out early next week. After that, we will have an image generated at `ceph/ceph:v14`, which we could use as the operator base image. At that point, this issue just go way. What if we go ahead and merge as is? In the interim few days, to run nautilus, the workaround would simply be to build with a different base image.

Reply:
That sounds acceptable to me, merging now will also probably help @jtlayton with some of his integration too. I have a couple of followups to do anyway. Thanks!